petaio.c

```diff
@@ -189,7 +218,9 @@ void petaio_read_internal(char * fname, int ic) {
         endrun(0, "Failed to close block: %s\n",
                     big_file_get_error_message());
     }
-    allocate_memory(NTotal[0] > 0);
+
+    /*Allocate the particle memory*/
+    petaio_alloc_particle_memory();
```

Comment:
I see the old code was doing this too. But when I look at this, It is seems to be clearer if the memory is allocated before read, since the read function doesn't return the allocated pointer as a return value; perhaps it is was impossible to do this previously -- the read_header function (which results All.MaxPart) was decoupled from read_internal only recently.